hazelcast/src/main/java/com/hazelcast/query/impl/IndexCopyBehavior.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.query.impl;
+
+/**
+ * Defines the behavior for index copying on index read/write.
+ */
+public enum IndexCopyBehavior {
```

Comment:
For each enums below, can we add some comments to address below questions?
- When to use this behaviour? 
- What performance effect can be expected?
- Can be used with `NATIVE` in memory format?

Reply:
Done 👍

Reply:
Also added to the GroupProperty javadoc.